nvflare/client/sub_process/process_api.py

```diff
@@ -0,0 +1,203 @@
+                )
+                flare_agent.start()
+
+            self.process_model_registry = ModelRegistry(client_config, rank, flare_agent)
+        except Exception as e:
+            print(f"flare.init failed: {e}")
+            raise e
+
+    def receive(self, timeout: Optional[float] = None) -> Optional[FLModel]:
+        """Receives model from NVFlare side.
+
+        Returns:
+            An FLModel received.
+        """
+        model_registry = self.get_model_registry()
+        return model_registry.get_model(timeout)
+
+    def send(self, model: FLModel, clear_cache: bool = True) -> None:
+        """Sends the model to Controller side.
+        Args:
+            model (FLModel): Sends a FLModel object.
+            clear_cache (bool): To clear the cache or not.
+        """
+        model_registry = self.get_model_registry()
+        model_registry.submit_model(model=model)
+        if clear_cache:
+            self.clear()
+
+    def system_info(self) -> Dict:
+        """Gets NVFlare system information.
+
+        System information will be available after a valid FLModel is received.
+        It does not retrieve information actively.
+
+        Returns:
+           A dict of system information.
+        """
+        model_registry = self.get_model_registry()
+        return model_registry.get_sys_info()
+
+    def get_config(self) -> Dict:
+        model_registry = self.get_model_registry()
+        return model_registry.config.config
+
+    def get_job_id(self) -> str:
+        sys_info = self.system_info()
+        return sys_info.get(FLMetaKey.JOB_ID, "")
+
+    def get_site_name(self) -> str:
+        sys_info = self.system_info()
+        return sys_info.get(FLMetaKey.SITE_NAME, "")
+
+    def get_task_name(self) -> str:
+        model_registry = self.get_model_registry()
+        return model_registry.get_task().task_name
+
+    def is_running(self) -> bool:
+        try:
+            self.receive()
+            return True
+        except FlareAgentException:
+            return False
+
+    def is_train(self) -> bool:
+        model_registry = self.get_model_registry()
+        if model_registry.rank != "0":
+            raise RuntimeError("only rank 0 can call is_train!")
+        return model_registry.task_name == model_registry.config.get_train_task()
+
+    def is_evaluate(self) -> bool:
+        model_registry = self.get_model_registry()
+        if model_registry.rank != "0":
+            raise RuntimeError("only rank 0 can call is_evaluate!")
+        return model_registry.task_name == model_registry.config.get_eval_task()
+
+    def is_submit_model(self) -> bool:
+        model_registry = self.get_model_registry()
+        if model_registry.rank != "0":
+            raise RuntimeError("only rank 0 can call is_submit_model!")
+        return model_registry.task_name == model_registry.config.get_submit_model_task()
+
+    def log(self, key: str, value: Any, data_type: AnalyticsDataType, **kwargs):
```

Comment:
make sure to copy all the docstrings from: https://github.com/NVIDIA/NVFlare/blob/main/nvflare/client/api.py